File: 2024/day_1/python/day_one.py
def get_diff(a: int, b: int):
    return abs(a - b)

def calcuate_distances(one, two):
    count = 0
    one.sort()
    two.sort()
    for index in range(len(one)):
        count += abs(one[index] - two[index])

    return count

def main():
    one = []
    two = []
    with open('input.txt', 'r') as file:
        for line in file:
            first, second = line.split()
            one.append(int(first))
            two.append(int(second))

    print(calcuate_distances(one, two))



if __name__ == "__main__":
    main()

File: 2024/day_1/typescript/day_one.ts
import * as fs from 'fs'

export const calculate_distance = (one: number[], two: number[]): number => {
  const first: number[] = one.sort((a, b) => a - b);
  const second: number[] = two.sort((a, b) => a - b);

  const count: number = first.reduce((total, value, index) => {
    const calculation: number = Math.abs(value - second[index])
    const new_total: number = total + calculation
    return new_total
  }, 0)


  return count;
};

const read_file = (data: string): number[][] => {
  const one: number[] = [];
  const two: number[] = [];
  data.split('\n').filter(val => val.trim() !== '').forEach((value) => {
    const arr = value.split(/\s+/).map((value) => { return parseInt(value.trim()) });
    one.push(arr[0])
    two.push(arr[1])
  })

  return [one, two]
}

function main() {
  const data: string = fs.readFileSync('input.txt', 'utf8')

  const arrays: number[][] = read_file(data)

  const total: number = calculate_distance(arrays[0], arrays[1])

  console.log(total)
}

if (require.main === module) {
  main();
}

